httpclient5/src/main/java/org/apache/hc/client5/http/impl/auth/RFC5987Codec.java

```diff
@@ -0,0 +1,155 @@
+/*
+ * ====================================================================
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ * ====================================================================
+ *
+ * This software consists of voluntary contributions made by many
+ * individuals on behalf of the Apache Software Foundation.  For more
+ * information on the Apache Software Foundation, please see
+ * <http://www.apache.org/>.
+ *
+ */
+
+package org.apache.hc.client5.http.impl.auth;
+
+import java.nio.ByteBuffer;
+import java.nio.CharBuffer;
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+import java.util.BitSet;
+
+import org.conscrypt.Internal;
+
+/**
+ * Utility class for encoding and decoding strings according to RFC 5987.
+ * This class provides methods to percent-encode and decode strings, particularly
+ * useful for handling HTTP header parameters that include non-ASCII characters.
+ *
+ * @Internal This class is intended for internal use within the library and
+ * should not be used as part of the public API.
+ */
+@Internal
+class RFC5987Codec {
```

Comment:
@arturobernalg There is a similar implementation in the `HttpRFC7578Multipart` class. Could you please get rid of it in this or a follow-up change-set?